// react-triangle/lib/support.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// eslint-disable-line no-unused-vars

var TriangleSupport = function (_Component) {
  (0, _inherits3.default)(TriangleSupport, _Component);

  function TriangleSupport() {
    (0, _classCallCheck3.default)(this, TriangleSupport);
    return (0, _possibleConstructorReturn3.default)(this, (TriangleSupport.__proto__ || (0, _getPrototypeOf2.default)(TriangleSupport)).apply(this, arguments));
  }

  (0, _createClass3.default)(TriangleSupport, [{
    key: 'render',
    value: function render() {
      var points = this._calcVertexCoord().map(function (s) {
        return s.join(',');
      }).join(' ');
      return _react2.default.createElement('polygon', (0, _extends3.default)({ points: points }, this.props));
    }
  }]);
  return TriangleSupport;
}(_react.Component);

exports.default = TriangleSupport;


TriangleSupport.propTypes = {
  x: _react.PropTypes.number,
  y: _react.PropTypes.number,
  direction: _react.PropTypes.oneOf(['up', 'down', 'left', 'right']),
  size: _react.PropTypes.number
};

TriangleSupport.defaultProps = {
  x: 0,
  y: 0,
  direction: 'up',
  size: 100
};
